File: go/internal/desugarer/simplify_lookups.go
package desugarer

import "github.com/10gen/mongoast/ast"

// getSimplifyLookups accepts a string representing the current database name and returns a pass
// that will remove the FromDB name from any $lookup when the FromDB name matches the current
// database name because that information is redudant and some implementations of mongodb aggregation
// language, e.g., mongod itself, do not support the FromDB field.
func getSimplifyLookups(currentDB string) func(pipeline *ast.Pipeline, _ uint64) *ast.Pipeline {
	return func(pipeline *ast.Pipeline, _ uint64) *ast.Pipeline {
		// modify the pipeline in place.
		for i := range pipeline.Stages {
			switch s := pipeline.Stages[i].(type) {
			case *ast.LookupStage:
				if s.FromDB == currentDB {
					s.FromDB = ""
				}
			}
		}

		return pipeline
	}
}

File: mongosql/src/air/optimizer/simplify_lookups.rs
use crate::air::optimizer::Pass;

/// Accepts a string representing the current database and returns a Pass that
/// will remove the from_db name from any Lookup when the from_db matches the
/// current database name. This is because the information is redundant and
/// some implementions of mongodb aggregation language, e.g. mongod itself, do
/// not support the from_db field.
pub fn get_simplify_lookups(_current_db: &str) -> Box<dyn Pass> {
    todo!()
}
